src/tests/stress/pedsim/Ped_util/Regions.enc

```diff
@@ -0,0 +1,462 @@
+  var top_right : Maybe[Box]
+  var top_left : Maybe[Box]
+  var bottom_left : Maybe[Box]
+  var bottom_right : Maybe[Box]
+  var matrix : Quad_tree
+  var agents : Maybe[Item]
+  var last : Maybe[Item]
+  var newcommers : Maybe[Item]
+  var ttl : int
+  var size : int
+  var moving : bool
+  def init(in_max : (int, int), in_min : (int, int)) : unit
+    this.agents=Nothing
+    this.last=Nothing
+    this.newcommers=Nothing
+    this.up=Nothing
+    this.down=Nothing
+    this.left=Nothing
+    this.right=Nothing
+    this.top_left=Nothing
+    this.bottom_right=Nothing
+    this.bottom_left=Nothing
+    this.matrix = new Quad_tree(in_max, in_min)
+    this.top_right=Nothing
+    match in_max with
+      case (x, y) =>
+        this.xmax = x
+        this.ymax = y
+      end
+    end
+    match in_min with
+      case (x, y) =>
+        this.xmin = x
+        this.ymin = y
+      end
+    end
+    this.ttl = 10001
+    this.size = 0
+    this.moving = false
+  end
+  def min() : (int, int)
+    (this.xmin, this.ymin)
+  end
+  def max() : (int, int)
+    (this.xmax, this.ymax)
+  end
+
+  def agents() : [(int, int)]
+    this.merge()
+    match this.agents with
+        case Nothing => new [(int, int)](0)
+        case thing =>
+              var cur=thing
+              var i = 0
+              while cur != Nothing do
+                i = i + 1
+                match cur with
+                    case Just(thing) => cur=thing.next
+                end
+              end
+              var ret = new [(int, int)](i)
+              cur = this.agents
+              i = 0
+              while cur != Nothing do
+                match cur with
+                    case Just(thing) =>
+                     cur=thing.next
+                     ret(i) = thing.a.pos()
+                    end
+                end
+                i = i + 1
+              end
+              ret
+        end
+    end
+  end
+  def add(a : Agent) : bool
+    if a.x > this.xmax || a.x < this.xmin || a.y > this.ymax || a.y < this.ymin then
+      false
+    else
+      this.add_internal(a,a.x,a.y)
+    end
+  end
+  def external_move(var a : Agent, x : int, y : int) : Maybe[Agent]
+    if x > this.xmax || x < this.xmin || y > this.ymax || y < this.ymin then
+      Just(a)
+    else
+      if this.matrix.get_val(x, y) == true then
+        Just(a)
+      else
+        a.move_int(x, y)
+        this.add_internal(a,x,y)
+        Nothing
+      end
+    end
+  end
+  def add_internal(a : Agent, x : int, y : int) : bool
+    if this.matrix.get_val(x,y) == true then
+      print("AGENTS IN TOP OF eachother\n")
+      while true do
+        ()
+      end
+      true
+    else
+      this.matrix.set(x,y)
+      if this.newcommers == Nothing then
+        this.newcommers = Just(new Item(a))
+      else
+        if this.last == Nothing then
+          this.last = Just(new Item(a))
+          match this.newcommers with
+              case Just(thing) => thing.append(this.last)
```

Comment:
could you give a more descriptive name to `thing`? I have no clue of what is represents